lib/node/closure-compiler-js.js

```diff
@@ -46,15 +46,20 @@ class CompilerJS {
 
   /**
    * @param {!Array<!{src: string, path: string, sourceMap: string}>} fileList
-   * @param {function(number, string, string)=} callback
+   * @param {function(number, Array<{src: string, path: string, sourceMap: (string|undefined)}>, string)=} callback
    * @return {child_process.ChildProcess}
    */
   run(fileList, callback) {
     const out = jscomp(this.flags, fileList);
+    out.warnings = [].slice.call(out.warnings);
```

Comment:
Could you add a comment explaining why you need to make fresh copies of the errors and warnings?